File: mindbell-instrumentationtest/src/main/java/de/dknapps/mindbell/test/MindBellTest.java
/**
 *
 */
package de.dknapps.mindbell.test;

import android.app.Activity;
import android.test.ActivityInstrumentationTestCase2;
import android.view.View;
import de.dknapps.mindbell.MindBell;

/**
 * @author marc
 *
 */
public class MindBellTest extends ActivityInstrumentationTestCase2<MindBell> {

    private Activity mActivity;
    private View mView;

    public MindBellTest() {
        super("de.dknapps.mindbell", MindBell.class);
    }

    @Override
    protected void setUp() throws Exception {
        super.setUp();
        mActivity = getActivity();
        mView = mActivity.findViewById(de.dknapps.mindbell.R.id.bell);

    }

    public void testPreconditions() {
        assertNotNull(mView);
    }
}

File: app/src/androidTest/java/com/googlecode/mindbell/MindBellTest.kt
/*
 * MindBell - Aims to give you a support for staying mindful in a busy life -
 *            for remembering what really counts
 *
 *     Copyright (C) 2010-2014 Marc Schroeder
 *     Copyright (C) 2014-2017 Uwe Damken
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.googlecode.mindbell

import android.app.Activity
import android.support.test.InstrumentationRegistry
import android.support.test.runner.AndroidJUnit4
import android.test.ActivityInstrumentationTestCase2
import android.test.suitebuilder.annotation.SmallTest
import android.view.View
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class MindBellTest : ActivityInstrumentationTestCase2<MindBell>(MindBell::class.java) {

    private var mActivity: Activity? = null
    private var mView: View? = null

    @Before
    @Throws(Exception::class)
    public override fun setUp() {
        super.setUp()
        // Injecting the Instrumentation instance is required for your test to run with AndroidJUnitRunner.
        injectInstrumentation(InstrumentationRegistry.getInstrumentation())
        mActivity = activity
        mView = mActivity!!.findViewById(com.googlecode.mindbell.R.id.bell)

    }

    @Test
    fun tesxtPreconditions() {
        Assert.assertNotNull(mView)
    }
}
